immunopepper/traversal.py

```diff
@@ -305,7 +306,7 @@ def get_and_write_peptide_and_kmer(peptide_dict=None, kmer_dict=None,
                         edge_expr = np.nan
 
                     ### Peptides
-                    add_peptide_properties(peptide_dict, [OutputMetadata(peptide=peptide.mut[pep_idx],
```

Comment:
This function is the ugly function collapsing the named tuples to get unicity on kmers. It was introduce to quickly reduce the io imprint of the software at the time when optimisation was performed. I am removing it on the foreground first and will test the speed of the software.

Reply:
See summary below explaining broader context: